### PR TITLE
Add aten::special_{i0e, i1, i1e, ndtri, log_ndtr, polynomial} OPs

### PR DESCRIPTION
**Pull request by LuFinch**
- [x] special_hermite_polynomial_he
- [x] special_hermite_polynomial_h
- [x] special_laguerre_polynomial_l
- [x] special_legendre_polynomial_p
- [x] special_i0e
- [x] special_i1e
- [x] special_i1
- [x] special_log_ndtr
- [x] special_ndtri

### CODE REVIEW

**Comment by xytintel** [comment]
@LuFinch Pls modify the filename. (e.g. `sycl/Hermite_polynomial_he.cpp` -> `sycl/HermitePolynomialHeKernel.cpp`)

File: src/ATen/native/xpu/sycl/HermitePolynomialHKernel.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/BinaryOps.h>
+#include <ATen/native/Math.h>
+#include <ATen/native/xpu/sycl/HermitePolynomialHKernel.h>
+#include <ATen/native/xpu/sycl/Loops.h>
+
+namespace at::native::xpu {
+
+template <typename scalar_t>
+struct hermite_polynomial_h_functor {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/HermitePolynomialHKernel.cpp:10]: Pls use `HermitePolynomialHFunctor`

File: src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp [hunk 2]
```diff
@@ -176,4 +176,73 @@ void logit_kernel(TensorIteratorBase& iter, const Scalar& eps_scalar) {
...
+
+template <typename scalar_t>
+struct I1Functor {
+  scalar_t operator()(scalar_t a) const {
+    return calc_i1(a);
+  }
+};
+
+void i1_kernel(TensorIteratorBase& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.common_dtype(), "i1_xpu", [&]() {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp:204]: Pls use `AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.common_dtype()`

File: src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp [hunk 3]
```diff
@@ -176,4 +176,73 @@ void logit_kernel(TensorIteratorBase& iter, const Scalar& eps_scalar) {
...
+
+template <typename scalar_t>
+struct I1eFunctor {
+  scalar_t operator()(scalar_t a) const {
+    return calc_i1e(a);
+  }
+};
+
+void i1e_kernel(TensorIteratorBase& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.common_dtype(), "i1e_xpu", [&]() {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp:217]: Pls use `AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.common_dtype(),`

File: src/ATen/native/xpu/sycl/HermitePolynomialHeKernel.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/BinaryOps.h>
+#include <ATen/native/Math.h>
+#include <ATen/native/xpu/sycl/HermitePolynomialHeKernel.h>
+#include <ATen/native/xpu/sycl/Loops.h>
+
+namespace at::native::xpu {
+
+template <typename scalar_t>
+struct hermite_polynomial_he_functor {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/HermitePolynomialHeKernel.cpp:10]: ditto.

File: src/ATen/native/xpu/sycl/LaguerrePolynomialLKernel.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/BinaryOps.h>
+#include <ATen/native/Math.h>
+#include <ATen/native/xpu/sycl/LaguerrePolynomialLKernel.h>
+#include <ATen/native/xpu/sycl/Loops.h>
+
+namespace at::native::xpu {
+
+template <typename scalar_t>
+struct laguerre_polynomial_l_functor {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/LaguerrePolynomialLKernel.cpp:10]: ditto.

File: src/ATen/native/xpu/sycl/LegendrePolynomialPKernel.cpp [hunk 6]
```diff
@@ -0,0 +1,55 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/BinaryOps.h>
+#include <ATen/native/Math.h>
+#include <ATen/native/xpu/sycl/LegendrePolynomialPKernel.h>
+#include <ATen/native/xpu/sycl/Loops.h>
+
+namespace at::native::xpu {
+
+template <typename scalar_t>
+struct legendre_polynomial_p_functor {
```
Comment (xytintel) [inline src/ATen/native/xpu/sycl/LegendrePolynomialPKernel.cpp:10]: ditto.

File: src/ATen/native/xpu/sycl/HermitePolynomialHKernel.cpp (hunk 1 above)
Comment (LuFinch) [reply src/ATen/native/xpu/sycl/HermitePolynomialHKernel.cpp]: Done

File: src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp (hunk 3 above)
Comment (LuFinch) [reply src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp]: Done

**Review by xytintel — APPROVED** [review]

**Comment by EikanWang** [comment]
Pls. help provide test case evidence.

**Comment by LuFinch** [comment]
Copy part of log from pull `/preci-ut/Torch-XPU-UT-Tests`. All other cases releated to this PR can pass.

```
2024-10-24T06:11:11.3652902Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_hermite_polynomial_h_xpu PASSED [  8%]
2024-10-24T06:11:11.4028181Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_hermite_polynomial_he_xpu PASSED [  8%]
2024-10-24T06:11:11.4215663Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_i0e_xpu PASSED       [  8%]
2024-10-24T06:11:11.4406891Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_i1_xpu PASSED        [  8%]
2024-10-24T06:11:11.4590297Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_i1e_xpu PASSED       [  8%]
2024-10-24T06:11:11.4973412Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_laguerre_polynomial_l_xpu PASSED [  8%]
2024-10-24T06:11:11.4979375Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_legendre_polynomial_p_xpu SKIPPED [  8%]
2024-10-24T06:11:11.5316997Z test_ops_xpu.py::TestCommonXPU::test_dtypes_special_log_ndtr_xpu PASSED  [  8%]
2024-10-24T06:11:19.7434737Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_hermite_polynomial_h_xpu_bool PASSED [ 18%]
2024-10-24T06:11:19.7523585Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_hermite_polynomial_he_xpu_bool PASSED [ 18%]
2024-10-24T06:11:19.7559269Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_i0e_xpu_bool PASSED [ 18%]
2024-10-24T06:11:19.7599595Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_i1_xpu_bool PASSED [ 18%]
2024-10-24T06:11:19.7635309Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_i1e_xpu_bool PASSED [ 18%]
2024-10-24T06:11:19.7726125Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_laguerre_polynomial_l_xpu_bool PASSED [ 18%]
2024-10-24T06:11:19.7733345Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_legendre_polynomial_p_xpu_bool SKIPPED [ 18%]
2024-10-24T06:11:19.7774764Z test_ops_xpu.py::TestCommonXPU::test_non_standard_bool_values_special_log_ndtr_xpu_bool PASSED [ 18%]
2024-10-24T06:11:53.4370375Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_hermite_polynomial_h_xpu_float32 PASSED [ 26%]
2024-10-24T06:11:53.4451427Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_hermite_polynomial_h_xpu_int64 PASSED [ 26%]
2024-10-24T06:11:53.4535834Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_hermite_polynomial_he_xpu_float32 PASSED [ 26%]
2024-10-24T06:11:53.4615835Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_hermite_polynomial_he_xpu_int64 PASSED [ 26%]
2024-10-24T06:11:53.4671267Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_i0e_xpu_float32 PASSED [ 26%]
2024-10-24T06:11:53.4711989Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_i0e_xpu_int64 PASSED [ 26%]
2024-10-24T06:11:53.4778705Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_i1_xpu_float32 PASSED [ 26%]
2024-10-24T06:11:53.4816292Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_i1_xpu_int64 PASSED [ 26%]
2024-10-24T06:11:53.4881737Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_i1e_xpu_float32 PASSED [ 26%]
2024-10-24T06:11:53.4919326Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_i1e_xpu_int64 PASSED [ 26%]
2024-10-24T06:11:53.5004811Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_laguerre_polynomial_l_xpu_float32 PASSED [ 26%]
2024-10-24T06:11:53.5085358Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_laguerre_polynomial_l_xpu_int64 PASSED [ 26%]
2024-10-24T06:11:53.5092580Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_legendre_polynomial_p_xpu_float32 SKIPPED [ 26%]
2024-10-24T06:11:53.5099026Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_legendre_polynomial_p_xpu_int64 SKIPPED [ 26%]
2024-10-24T06:11:53.5162638Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_log_ndtr_xpu_float32 PASSED [ 26%]
2024-10-24T06:11:53.5204238Z test_ops_xpu.py::TestCommonXPU::test_noncontiguous_samples_special_log_ndtr_xpu_int64 PASSED [ 26%]
2024-10-24T06:12:15.8440628Z test_ops_xpu.py::TestCommonXPU::test_out_special_hermite_polynomial_h_xpu_float32 PASSED [ 36%]
2024-10-24T06:12:15.8582718Z test_ops_xpu.py::TestCommonXPU::test_out_special_hermite_polynomial_he_xpu_float32 PASSED [ 36%]
2024-10-24T06:12:15.8631082Z test_ops_xpu.py::TestCommonXPU::test_out_special_i0e_xpu_float32 PASSED  [ 36%]
2024-10-24T06:12:15.8678355Z test_ops_xpu.py::TestCommonXPU::test_out_special_i1_xpu_float32 PASSED   [ 36%]
2024-10-24T06:12:15.8725716Z test_ops_xpu.py::TestCommonXPU::test_out_special_i1e_xpu_float32 PASSED  [ 36%]
2024-10-24T06:12:15.8870183Z test_ops_xpu.py::TestCommonXPU::test_out_special_laguerre_polynomial_l_xpu_float32 PASSED [ 36%]
2024-10-24T06:12:15.8875786Z test_ops_xpu.py::TestCommonXPU::test_out_special_legendre_polynomial_p_xpu_float32 SKIPPED [ 36%]
2024-10-24T06:12:15.8931645Z test_ops_xpu.py::TestCommonXPU::test_out_special_log_ndtr_xpu_float32 PASSED [ 36%]
2024-10-24T06:12:40.0638697Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_hermite_polynomial_h_xpu PASSED [ 41%]
2024-10-24T06:12:40.0909441Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_hermite_polynomial_he_xpu PASSED [ 41%]
2024-10-24T06:12:40.0997163Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_i0e_xpu PASSED  [ 41%]
2024-10-24T06:12:40.1079479Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_i1_xpu PASSED   [ 41%]
2024-10-24T06:12:40.1162678Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_i1e_xpu PASSED  [ 41%]
2024-10-24T06:12:40.1439267Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_laguerre_polynomial_l_xpu PASSED [ 41%]
2024-10-24T06:12:40.1444863Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_legendre_polynomial_p_xpu SKIPPED [ 41%]
2024-10-24T06:12:40.1533390Z test_ops_xpu.py::TestCommonXPU::test_out_warning_special_log_ndtr_xpu PASSED [ 41%]
```

**Pull request merged**